### PR TITLE
MINOR: Code refacotring in KTable-KTable Join

### PR DESCRIPTION
**Pull request by guozhangwang**
1. Rename KTableKTableJoin to KTableKTableInnerJoin. Also removed `abstract` from other joins.
2. Merge KTableKTableJoinValueGetter.java into KTableKTableInnerJoin.
3. Use set instead of arrays in the stores function, to avoid duplicate stores to be connected to processors.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@bbejeck

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java [hunk 1]
```diff
@@ -100,4 +97,38 @@ public void process(final K key, final Change<V1> change) {
...
+        private final KTableValueGetter<K, V2> valueGetter2;
+
+        KTableKTableInnerJoinValueGetter(final KTableValueGetter<K, V1> valueGetter1,
+                                         final KTableValueGetter<K, V2> valueGetter2) {
+            this.valueGetter1 = valueGetter1;
+            this.valueGetter2 = valueGetter2;
+        }
+
+        @Override
+        public void init(ProcessorContext context) {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java]: nit: add `final`

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java [hunk 2]
```diff
@@ -100,4 +97,38 @@ public void process(final K key, final Change<V1> change) {
...
+            valueGetter1.init(context);
+            valueGetter2.init(context);
+        }
+
+        @Override
+        public R get(K key) {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java]: nit: add `final`

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java [hunk 3]
```diff
@@ -100,4 +97,38 @@ public void process(final K key, final Change<V1> change) {
...
+            R newValue = null;
+            V1 value1 = valueGetter1.get(key);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java]: nit: add `final`

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java [hunk 4]
```diff
@@ -100,4 +97,38 @@ public void process(final K key, final Change<V1> change) {
...
+
+        @Override
+        public void init(ProcessorContext context) {
+            valueGetter1.init(context);
+            valueGetter2.init(context);
+        }
+
+        @Override
+        public R get(K key) {
+            R newValue = null;
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java]: Can we remove `newValue` and do `return null` and `return joiner.apply(value1, value2);` instead?

**Review by mjsax — APPROVED** [review]
Just some nits. Feel free to merge.

**Comment by bbejeck** [comment]
no additional comment beyond @mjsax, LGTM

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java (hunk 4 above)
Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java]: ack

**Pull request merged**